### DISCUSSION TITLE
Best practices for polling for data

### DISCUSSION THREAD

**Discussion by APTy** [lowdefy/lowdefy#657] [Questions and Help] [answered]
Has anyone had to poll for data yet? e.g. `make a GET request every 5 seconds`. I'm curious if there's some prior art on it.

I was considering having an invisible button that would make the request that updates the page, then use a `JsAction` with a `setInterval()` to trigger the button periodically. Thoughts?

**Comment by Gervwyk** [comment, marked as answer]
This would work, but we can probably come up with a more elegant solution, something like a polling setting on requests would be great. We'll have a look at this when implementing #640 

For triggering a button click, I think this is a bit of a hack that can work for now. Just remember that your button `onClick` the `Request` action to refetch the data.

We usually just fetch the data onEnter. Currently not really had a case where we need to poll the data for a "live" dashboard. But we have promised a customer a live dashboard 🙈 So the feature is definitely on the roadmap.

**Reply by APTy** [reply]
A polling setting on requests would be a nice touch, agreed! Plus users could use that to set up conditionals for how/when to poll and when to stop 😁

**Reply by Gervwyk** [reply]
True. Could actually be really easy to implement if we do it as a param on the `Request` action. So the polling is set up the first time the action is triggered. Also since the params are parsed, you could then use any operator logic to control polling.

**Reply by Gervwyk** [reply]
I've created a feature issue #660 for this 🎈

**Reply by Gervwyk** [reply]
I think this could make sense as a general action property instead of only for `Request`. Like triggering a `JsAction` on an interval will also be useful.
